notebooks/Streaming_Dataset.ipynb

```diff
@@ -0,0 +1,194 @@
+    "        self.fc1 = nn.Linear(input_dim, hidden_dim)\n",
+    "        self.fc2 = nn.Linear(hidden_dim, output_dim)\n",
+    "\n",
+    "    def forward(self, X):\n",
+    "        X = F.relu(self.fc1(X))\n",
+    "        return self.fc2(X)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Custom Callback\n",
+    "\n",
+    "We will now setup a validation callback for use during model training. The callback is defined to evaluate the model’s accuracy on a validation set at the end of each training epoch. The training dataset and  validation dataset are wrapped in Torch `DataLoader` to enable batch processing."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Create streaming datasets\n",
+    "train_ds = StreamingDataset(length=1000, seed=0)\n",
+    "valid_ds = StreamingDataset(length=200, seed=1)\n",
+    "\n",
+    "train_loader = DataLoader(train_ds, batch_size=16)\n",
+    "valid_loader = DataLoader(valid_ds, batch_size=16)\n",
+    "\n",
+    "class StreamingValidationCallback(Callback):\n",
+    "    def __init__(self, valid_loader, name='valid_acc'):\n",
+    "        self.valid_loader = valid_loader\n",
+    "        self.name = name\n",
+    "    \n",
+    "    def on_epoch_end(self, net, dataset_train, dataset_valid, **kwargs):\n",
+    "        net.module_.eval()\n",
+    "        correct = 0\n",
+    "        total = 0\n",
+    "        \n",
+    "        with torch.no_grad():\n",
+    "            for X_batch, y_batch in self.valid_loader:\n",
+    "                X_batch = X_batch.to(net.device)\n",
+    "                if isinstance(y_batch, (list, tuple)):\n",
+    "                    y_batch = torch.tensor(y_batch).to(net.device)\n",
+    "                else:\n",
+    "                    y_batch = y_batch.to(net.device)\n",
+    "                \n",
+    "                outputs = net.module_(X_batch)\n",
+    "                _, predicted = torch.max(outputs.data, 1)\n",
+    "                total += y_batch.size(0)\n",
+    "                correct += (predicted == y_batch).sum().item()\n",
+    "        \n",
+    "        accuracy = correct / total\n",
+    "        print(f\"  {self.name}: {accuracy:.4f}\")\n",
+    "        net.history.record(self.name, accuracy)\n",
+    "\n",
+    "valid_ds_for_callback = StreamingDataset(length=200, seed=1)\n",
+    "valid_loader_for_callback = DataLoader(valid_ds_for_callback, batch_size=16)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Model Training and Result\n",
+    "\n",
+    "We will now wrap the neural net classifier with the torch model and use `DataLoader` to create data batches. We will also add the custom validation callback seperately from the training loop. It is important to note that since we are using `IterableDataset`,we will set the train_split to None.\n",
```

Comment:
Let's add a short explanation.

```suggestion
    "We will now wrap the neural net classifier with the torch model and use `DataLoader` to create data batches. We will also add the custom validation callback separately from the training loop. It is important to note that since we are using `IterableDataset`, we have to set the `train_split` to `None`, since streaming data cannot be split in advance.\n",
```